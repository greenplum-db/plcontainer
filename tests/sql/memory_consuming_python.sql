-- memory consuming tests for python
CREATE OR REPLACE FUNCTION py_memory_allocate_c(num int) RETURNS void AS $$
# container: plc_python_shared
allocate = 'a' * num * 1024 * 1024
$$ LANGUAGE plcontainer;


CREATE OR REPLACE FUNCTION py_memory_allocate_return_c(num int) RETURNS text[] AS $$
# container: plc_python_shared
allocate = 'a' * num * 1024 * 1024
return ['Allocate:' + str(num) + 'MB as ' + allocate[1024*1024*(num/2)]]
$$ LANGUAGE plcontainer;

select py_memory_allocate_c(128);
select py_memory_allocate_c(128);
select py_memory_allocate_c(128);
select py_memory_allocate_c(128);
select py_memory_allocate_c(128);
select py_memory_allocate_c(128);
select py_memory_allocate_c(128);
select py_memory_allocate_c(128);
select py_memory_allocate_c(128);
select py_memory_allocate_c(128);

select py_memory_allocate_return_c(16);
select py_memory_allocate_return_c(32);
select py_memory_allocate_return_c(64);
select py_memory_allocate_return_c(128);
select py_memory_allocate_return_c(256);
select py_memory_allocate_return_c(256);
select py_memory_allocate_return_c(512);
select py_memory_allocate_return_c(1024);
