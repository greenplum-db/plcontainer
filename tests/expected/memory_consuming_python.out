-- memory consuming tests for python
CREATE OR REPLACE FUNCTION py_memory_allocate_c(num int) RETURNS void AS $$
# container: plc_python_shared
allocate = 'a' * num * 1024 * 1024
$$ LANGUAGE plcontainer;
CREATE OR REPLACE FUNCTION py_memory_allocate_return_c(num int) RETURNS text[] AS $$
# container: plc_python_shared
allocate = 'a' * num * 1024 * 1024
return ['Allocate:' + str(num) + 'MB as ' + allocate[1024*1024*(num/2)]]
$$ LANGUAGE plcontainer;
select py_memory_allocate_c(128);
 py_memory_allocate_c 
----------------------
 
(1 row)

select py_memory_allocate_c(128);
 py_memory_allocate_c 
----------------------
 
(1 row)

select py_memory_allocate_c(128);
 py_memory_allocate_c 
----------------------
 
(1 row)

select py_memory_allocate_c(128);
 py_memory_allocate_c 
----------------------
 
(1 row)

select py_memory_allocate_c(128);
 py_memory_allocate_c 
----------------------
 
(1 row)

select py_memory_allocate_c(128);
 py_memory_allocate_c 
----------------------
 
(1 row)

select py_memory_allocate_c(128);
 py_memory_allocate_c 
----------------------
 
(1 row)

select py_memory_allocate_c(128);
 py_memory_allocate_c 
----------------------
 
(1 row)

select py_memory_allocate_c(128);
 py_memory_allocate_c 
----------------------
 
(1 row)

select py_memory_allocate_c(128);
 py_memory_allocate_c 
----------------------
 
(1 row)

select py_memory_allocate_return_c(16);
 py_memory_allocate_return_c 
-----------------------------
 {"Allocate:16MB as a"}
(1 row)

select py_memory_allocate_return_c(32);
 py_memory_allocate_return_c 
-----------------------------
 {"Allocate:32MB as a"}
(1 row)

select py_memory_allocate_return_c(64);
 py_memory_allocate_return_c 
-----------------------------
 {"Allocate:64MB as a"}
(1 row)

select py_memory_allocate_return_c(128);
 py_memory_allocate_return_c 
-----------------------------
 {"Allocate:128MB as a"}
(1 row)

select py_memory_allocate_return_c(256);
 py_memory_allocate_return_c 
-----------------------------
 {"Allocate:256MB as a"}
(1 row)

select py_memory_allocate_return_c(256);
 py_memory_allocate_return_c 
-----------------------------
 {"Allocate:256MB as a"}
(1 row)

select py_memory_allocate_return_c(512);
 py_memory_allocate_return_c 
-----------------------------
 {"Allocate:512MB as a"}
(1 row)

select py_memory_allocate_return_c(1024);
 py_memory_allocate_return_c 
-----------------------------
 {"Allocate:1024MB as a"}
(1 row)

